docs/src/content/en/docs/agents/overview.mdx

```diff
@@ -54,264 +78,237 @@ export const mastra = new Mastra({
-  required: ["summary", "keywords"],
-};
+Define the `output` shape using [JSON Schema](https://json-schema.org/):
 
+```typescript showLineNumbers copy
 const response = await testAgent.generate(
   [
     {
-      role: "user",
-      content:
-        "Please provide a summary and keywords for the following text: ...",
+      role: "system",
+      content: "Provide a summary and keywords for the following text:"
     },
+    {
+      role: "user",
+      content: "Monkey, Ice Cream, Boat"
+    }
   ],
   {
-    output: schema,
-  },
+    output: {
+      type: "object",
+      properties: {
+        summary: { type: "string" },
+        keywords: { type: "array", items: { type: "string" } }
+      },
+      additionalProperties: false,
+      required: ["summary", "keywords"]
+    }
+  }
 );
 
-console.log("Structured Output:", response.object);
+console.log(response.object);
 ```
 
 ### Using Zod
 
-You can also use Zod schemas for type-safe structured outputs.
-
-First, install Zod:
-
-```bash npm2yarn copy
-npm install zod
-```
-
-Then, define a Zod schema and use it with the agent:
+Define the `output` shape using [Zod](https://zod.dev/):
 
 ```typescript showLineNumbers copy
 import { z } from "zod";
 
-// Define the Zod schema
-const schema = z.object({
-  summary: z.string(),
-  keywords: z.array(z.string()),
-});
-
-// Use the schema with the agent
 const response = await testAgent.generate(
   [
     {
-      role: "user",
-      content:
-        "Please provide a summary and keywords for the following text: ...",
+      role: "system",
+      content: "Provide a summary and keywords for the following text:"
     },
+    {
+      role: "user",
+      content: "Monkey, Ice Cream, Boat"
+    }
   ],
   {
-    output: schema,
-  },
+    output: z.object({
+      summary: z.string(),
+      keywords: z.array(z.string())
+    })
+  }
 );
 
-console.log("Structured Output:", response.object);
+console.log(response.object);
 ```
 
 ### Using Tools
 
-If you need to generate structured output alongside tool calls, you'll need to use the `experimental_output` property instead of `output`. Here's how:
+If you need to generate structured output with an agent that uses a tool, use `experimental_output` instead of `output`:
 
 ```typescript showLineNumbers copy
-const schema = z.object({
-  summary: z.string(),
-  keywords: z.array(z.string()),
-});
-
 const response = await testAgent.generate(
   [
     {
-      role: "user",
-      content:
-        "Please analyze this repository and provide a summary and keywords...",
+      role: "system",
+      content: "Provide a summary and keywords for the following text:"
     },
+    {
+      role: "user",
+      content: "Monkey, Ice Cream, Boat"
+    }
   ],
   {
-    // Use experimental_output to enable both structured output and tool calls
-    experimental_output: schema,
-  },
+    experimental_output: z.object({
+      summary: z.string(),
+      keywords: z.array(z.string())
+    })
+  }
 );
```

Comment:
For system messages we should show it more like this:


```suggestion
const response = await testAgent.generate("Monkey, Ice Cream, Boat",
{
    experimental_output: z.object({
      summary: z.string(),
      keywords: z.array(z.string())
    }),
    context: [
	    {
	      role: "system",
	      content: "Provide a summary and keywords for the following text:"
	    }
    ]
  }
);
```

Getting people to use the array syntax and add multiple input messages makes it more likely to cause unexpected behaviours when combined with memory

Reply:
but really we don't need a system message here

```ts
const response = await testAgent.generate("Provide a summary and keywords for the following text: Monkey, Ice Cream, Boat",
  {
    experimental_output: z.object({
      summary: z.string(),
      keywords: z.array(z.string())
    })
  }
);
```